forem/forem
# [2020 EOY Issue] Modal alert when enabling Auth Provider without keys

## What type of PR is this? (check all applicable)


## Description
This PR alerts admins via a modal when they attempt to enable Auth Providers without entering their respective key and secret. The PR also refactors the `adminModal.js` function to make it more generalizable for more use-cases.

## Related Tickets & Documents
This PR introduces an improved User Flow, that will replace the Warning Banner in [this PR](https://github.com/forem/forem/pull/11696).
This PR is a continuation of [this closed PR](https://github.com/forem/forem/pull/11750) which lives on my forked forem repo. I have since just cloned the forem repo directly.

## QA Instructions, Screenshots, Recordings
1. As `:super_admin` role, navigate to `localhost:xxxx/admin/config`. Toggle open the Authentication section.
2. Enable one or more Authentication Providers, but leave the keys blank (or you can type random string for Key OR Secret, but NOT both). Update Site Configuration. A modal should popup listing the providers you attempted to enable.
3. Click "Continue editing" button or "X" the modal. You should be taken back to the main Config page AND the Update Site Configuration button should NOT be disabled.
4. Run through items 1 and 2 again, but this time, click "Cancel". ~"Save anyway"~ The modal should close, the enabled-indicator should be gone from the providers you attempted to enable, and the providers' sections should be closed. ~The update action should fire, the page refresh, but the keyless provider(s) you attempted to enable should still be disabled.~
5. Re-enable your chosen providers, but this time, enter dummy strings in both key fields. Update site configuration. The page should refresh, and you should see your chosen providers are enabled.

Demo Video of PR:
(Updated Demo) https://www.loom.com/share/f31fe527a7434b559888712e873dcecb
(Old Demo) https://www.loom.com/share/2ffd7af973af4054b21832f59cb9eec9

### UI accessibility concerns?
None anticipated

## Added tests?

- [X] Will be done in a subsequent PR;  _waiting on a couple Cypress-related PRs from @nickytonline_

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [X] No documentation needed; docs already reflect the need to enter auth provider key and secret

## [optional] Are there any post deployment tasks we need to perform?
No

## [optional] What gif best describes this PR or how it makes you feel?

![The late great Aaliyah](https://media.giphy.com/media/3o6MbaH9LYlaVpHWBa/giphy.gif)
